control/control_loop.go

```diff
@@ -113,31 +114,40 @@ func createLoop(logger logging.Logger, cfg Config, m Controllable) (*Loop, error
 			l.activeBackgroundWorkers.Add(1)
 			utils.ManagedGo(func() {
 				b := b
-				nInputs := len(b.ins)
 				close(waitCh)
 				for {
-					sw := make([]*Signal, nInputs)
-					for i, c := range b.ins {
+					sw := []*Signal{}
+					s := []*Signal{}
+					for _, c := range b.ins {
 						r, ok := <-c
 						if !ok {
 							b.mu.Lock()
 							for _, out := range b.outs {
 								close(out)
 							}
-							// logger.Debugf("Closing outs for block %s %+v\r\n", b.blk.Config(ctx).Name, r)
 							b.outs = nil
 							b.mu.Unlock()
 							return
 						}
-						if len(r) == 1 {
-							sw[i] = r[0]
+						for j := 0; j < len(r); j++ {
+							if r[j] != nil {
+								sw = append(sw, r[j])
+							}
+						}
+						// TODO(npmenard) do we want to support multidimentional signals?
+						//nolint: makezero
+					}
+					if strings.Contains(b.blk.Config(l.cancelCtx).Name, "PID") {
+						if strings.Contains(b.blk.Config(l.cancelCtx).Name, "ang") {
+							s = append(s, sw[1])
```

Comment:
Let's make a follow-up ticket for multi-dimensional inputs and remove the hardcoding from the contrl_loop code. 

We're going to need to append the second signal in the same way we append the first, and I believe the logic will have to do with the number of ins and the number of outs.. Let's call the second variable `s` for the second signal, `tempSecondSignal` to remember that this code is the thing that we're dealing with later.